qiskit_nature/algorithms/initial_points/mp2_initial_point.py

```diff
@@ -0,0 +1,302 @@
+    point with a VQE algorithm alongside a UCC ansatz.
+
+    The MP2 calculation requires the two-body molecular orbital matrix, the orbital energies,
+    and the Hartree-Fock reference energy from the ``ElectronicEnergy`` in the grouped
+    property.
+
+    An MP2InitialPoint uses the same excitations as generated by the UCC ansatz to ensure that
+    the coefficients are mapped correctly in the initial point. Only initial point values that
+    correspond to double-excitations will be non-zero.
+
+    The coefficients and energy deltas are computed using the ``get_initial_point`` method.
+    Thereafter, they can be accessed via the ``initial_point`` and ``energy_deltas`` properties.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.threshold = 1e-12
+        self._corrections: dict[str, tuple[float, float]] = None
+
+    @property
+    def grouped_property(self) -> GroupedSecondQuantizedProperty:
+        """The grouped property."""
+        return self._grouped_property
+
+    @grouped_property.setter
+    def grouped_property(self, grouped_property: GroupedSecondQuantizedProperty) -> None:
+        """The grouped property.
+
+        The grouped property is required to contain the ElectronicEnergy, which must contain
+        the two-body molecular orbital matrix, the orbital energies, and the Hartree-Fock
+        reference energy.
+
+        Raises:
+            QiskitNatureError: If the ``ElectronicEnergy`` is missing or the two-body molecular
+            orbital matrix, the orbital energies, and/or the Hartree-Fock reference energy are
+            missing from ``ElectronicEnergy``.
+        """
+        try:
+            electronic_energy: ElectronicEnergy = grouped_property.get_property(ElectronicEnergy)
+        except TypeError as invalid_grouped_property:
+            raise QiskitNatureError from invalid_grouped_property
+
+        try:
+            self._integral_matrix: np.ndarray = electronic_energy.get_electronic_integral(
+                ElectronicBasis.MO, 2
+            ).get_matrix()
+            # Infer the number of molecular orbitals from the MO matrix.
+            self._num_molecular_orbitals: int = self._integral_matrix.shape[0]
+        except TypeError as missing_two_body_mo_matrix:
+            raise QiskitNatureError from missing_two_body_mo_matrix
+
+        self._orbital_energies: np.ndarray = electronic_energy.orbital_energies
+        if self._orbital_energies is None:
+            raise QiskitNatureError(
+                f"Orbital energies not found in grouped property: {grouped_property}"
+            )
+
+        self._reference_energy: float = electronic_energy.reference_energy
+        if self._reference_energy is None:
+            raise QiskitNatureError(
+                f"Hartree-Fock reference energy not found in grouped property: {grouped_property}"
+            )
```

Comment:
Is this energy strictly needed or is it purely optional in order to provide the correct total energy? Maybe you can log a warning rather than raise one and set the internal object to `float("NaN")`. (Note: this will cause `.energy` to also return `NaN`, but `.energy_delta` is still available so I'd say this is fine)

Reply:
So, previously I had `.energy` just return `.delta` if reference_energy was None. (i.e. just make the reference_energy 0). But then I thought the reference energy would always be zero. I can reinstate that logic (with a warning).